test/functions/vass/subclass-to-ref-formal.chpl

```diff
@@ -0,0 +1,14 @@
+// Test that for a '[const] ref' formal, the type of the actual
+// must equal the type of the formal.
+// (Unless an implicit conversion comes to play.)
+
+class C    { }
+class D: C { }
+
+proc test(ref lhs: C) {}
```

Comment:
Yeah, it is not intuitive. The reason it shouldn't work is because if we allowed it, the following program would compile yet produce undefined behavior at run time:

    class C    { }
    class D: C { proc print { writeln(); } }

    proc test(ref lhs: C) { lhs = new C(); }

    var d = new D();
    test(d);
    d.print;  // passes static checking, invokes non-existing C.print at runtime